swagger-codegen3: init at 3.0.25

1000101 opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

1000101 committed:
swagger-codegen3: init at 3.0.25

SuperSandro2000 commented on pkgs/tools/networking/swagger-codegen/default.nix:
```diff
@@ -11,8 +11,8 @@ stdenv.mkDerivation rec {
+    sha256 = "0akkc0mdxp2mhn8dl3lwv6y11b718aza78cndn9p7736ffw5q6s0";
   };
 
   phases = [ "installPhase" ];
```
Please do not overwrite phases and instead the one that should no run.

1000101 replied:
I'm sorry, could you please elaborate on this?

SuperSandro2000 replied:
```suggestion
```

and add ``dontBuild = false;`` etc.

1000101 replied:
Ah, sorry it took me so long, should be resolved now. Thanks!

SuperSandro2000 replied:
You messed something up while pushing this change.

1000101 replied:
I've also changed the whole structure and move the package into new directory (swagger-codegen3) since I realized we might need both the current `swagger-codegen` (for v2) and this one. I've also changed the output binary name (appended `3` at the end) so that it would be possible to have both installed concurrently.

1000101 replied:
What do you mean by messing it up? Is it related to ^^^ ?

SuperSandro2000 replied:
You did not change this line. phases should not be overwritten in leaf packages.

1000101 replied:
Ah, gosh, I feel stupid now. I didn't notice that I've left v2 here as well (it changed in the meantime but I didn't do a rebase). Sorry!!!

1000101 replied:
Should I create a new PR for the original `swagger-codegen` (to get rid of the `installPhase`) or would you like me to do it here?

SuperSandro2000 replied:
You can do that in this PR.

1000101 replied:
Ok, hopefully fixed. Thanks a million for your patience.

1000101 commented:
@SuperSandro2000 I've just found out that both versions 2.X.X and 3.X.X are currently supported and maintained for different schema versions. Therefore, I've moved this definition to a new package so that both `swagger-codegen` versions could be packaged and updated independently.

SuperSandro2000 commented on pkgs/tools/networking/swagger-codegen3/default.nix:
```diff
@@ -0,0 +1,33 @@
+    url = "https://repo1.maven.org/maven2/io/swagger/codegen/v3/${pname}-cli/${version}/${jarfilename}";
+    sha256 = "1rdz45kmmg60fs7ddnla1xq30nah6s6rd18fqbjbjxng8r92brnd";
+  };
+
+  unpackPhase = ":";
```
```suggestion

  dontUnpack = true;
```

1000101 replied:
Thanks!!!

SuperSandro2000 commented on pkgs/tools/networking/swagger-codegen3/default.nix:
```diff
@@ -0,0 +1,33 @@
+  dontUnpack = true;
+
+  installPhase = ''
+    install -D "$src" "$out/share/java/${jarfilename}"
```
```suggestion
    install -D $src $out/share/java/${jarfilename}
```

1000101 committed:
swagger-codegen: do not override phases

SuperSandro2000 commented:
Can you please put the changes into one commit per package? After that we can merge this.

1000101 commented:
> Can you please put the changes into one commit per package? After that we can merge this.

Wonderful, thank you so much for your help!!!
